### ISSUE TITLE
py2app error when launching on macos

### ISSUE THREAD

**Issue by benlear** [spyder-ide/spyder#20704]
## Problem Description
This appears to be an issue similar to: https://github.com/spyder-ide/spyder/issues/19236

### What steps reproduce the problem?

1. Simply launching Spyder produces the error. I have a M1 mac running Ventura 13.2.1

### What is the expected output? What do you see instead?
I expect spyder to launch successfully.  Instead, the app appears to load, and then I see a dialog window saying: Launch error: see the py2app wesite for debugging

I have this error even after reinstalling the lastest version of spyder.

**Comment by ccordoba12** [comment]
Hey @benlear, thanks for reporting. You said:

> I expect spyder to launch successfully. Instead, the app appears to load, and then I see a dialog window saying: Launch error: see the py2app wesite for debugging

Could you post a screenshot with the contents of that dialog? That way we can better understand what's happening in your case.

**Comment by benlear** [comment]
Absolutely, screenshot is attached. 
<img width="250" alt="Screenshot 2023-03-18 at 4 14 11 PM" src="https://user-images.githubusercontent.com/7529372/226193457-6c099083-5651-44ac-94ab-21b16c7785ec.png">

**Comment by ccordoba12** [comment]
Thanks @benlear!

@mrclary, could you take a look at this one? I think it's possible to start the app from the command line, so that we can take a look at the actual error, right?

**Comment by mrclary** [comment]
To launch the standalone application from the command-line, do the following.
```
$ /Applications/Spyder.app/Contents/MacOS/Spyder --debug-info=minimal
```
This should provide standard output of logs and error messaging to the terminal.

**Comment by benlear** [comment]
Thanks.  Here is the error output. I appreciate the help:

$ /Applications/Spyder.app/Contents/MacOS/Spyder --debug-info=minimal
The available OpenGL surface format was either not version 3.2 or higher or not a Core Profile.
Chromium on macOS will fall back to software rendering in this case.
Hardware acceleration and features such as WebGL will not be available.
fromIccProfile: failed minimal tag size sanity
2023-03-19 20:31:27,104 [INFO] [spyder.app.mainwindow] -> Start of MainWindow constructor
2023-03-19 20:31:27,105 [INFO] [spyder.app.mainwindow] -> End of MainWindow constructor
2023-03-19 20:31:27,105 [INFO] [spyder.app.mainwindow] -> *** Start of MainWindow setup ***
2023-03-19 20:31:27,105 [INFO] [spyder.app.mainwindow] -> Applying theme configuration...
2023-03-19 20:31:27,118 [INFO] [spyder.app.mainwindow] -> Loading switcher...
2023-03-19 20:31:28,875 [INFO] [spyder.app.mainwindow] -> Loading Appearance...
2023-03-19 20:31:28,888 [INFO] [spyder.app.mainwindow] -> Loading appearance...
2023-03-19 20:31:28,889 [INFO] [spyder.app.mainwindow] -> Registering shortcuts for appearance...
2023-03-19 20:31:28,902 [INFO] [spyder.app.mainwindow] -> Loading Application...
2023-03-19 20:31:28,905 [INFO] [spyder.app.mainwindow] -> Loading application...
2023-03-19 20:31:28,905 [INFO] [spyder.app.mainwindow] -> Registering shortcuts for application...
2023-03-19 20:31:28,920 [INFO] [spyder.app.mainwindow] -> Loading Breakpoints...
2023-03-19 20:31:28,922 [INFO] [spyder.app.mainwindow] -> Loading breakpoints...
2023-03-19 20:31:28,931 [INFO] [spyder.app.mainwindow] -> Registering shortcuts for breakpoints...
2023-03-19 20:31:29,001 [INFO] [spyder.app.mainwindow] -> Loading Completion and linting...
2023-03-19 20:31:29,003 [INFO] [spyder.app.mainwindow] -> Loading completions...
2023-03-19 20:31:29,003 [INFO] [spyder.app.mainwindow] -> Registering shortcuts for completions...
2023-03-19 20:31:29,032 [INFO] [spyder.app.mainwindow] -> Loading Files...
2023-03-19 20:31:29,083 [INFO] [spyder.app.mainwindow] -> Loading explorer...
2023-03-19 20:31:29,095 [INFO] [spyder.app.mainwindow] -> Registering shortcuts for explorer...
2023-03-19 20:31:29,119 [INFO] [spyder.app.mainwindow] -> Loading Find...
2023-03-19 20:31:29,123 [INFO] [spyder.app.mainwindow] -> Loading find_in_files...
2023-03-19 20:31:29,138 [INFO] [spyder.app.mainwindow] -> Registering shortcuts for find_in_files...
2023-03-19 20:31:29,268 [INFO] [spyder.app.mainwindow] -> Loading Help...
2023-03-19 20:31:29,272 [INFO] [spyder.app.mainwindow] -> Loading help...
2023-03-19 20:31:29,296 [INFO] [spyder.app.mainwindow] -> Registering shortcuts for help...
2023-03-19 20:31:29,456 [INFO] [spyder.app.mainwindow] -> Loading History...
2023-03-19 20:31:29,461 [INFO] [spyder.app.mainwindow] -> Loading historylog...
2023-03-19 20:31:29,878 [INFO] [spyder.app.mainwindow] -> Registering shortcuts for historylog...
2023-03-19 20:31:29,881 [INFO] [spyder.plugins.console.widgets.main_widget] -> Initializing...
2023-03-19 20:31:29,944 [INFO] [spyder.app.mainwindow] -> Loading Internal console...
2023-03-19 20:31:29,948 [INFO] [spyder.app.mainwindow] -> Loading internal_console...
2023-03-19 20:31:29,958 [INFO] [spyder.app.mainwindow] -> Registering shortcuts for internal_console...
2023-03-19 20:31:30,006 [INFO] [spyder.app.mainwindow] -> Loading IPython console...
2023-03-19 20:31:30,011 [INFO] [spyder.app.mainwindow] -> Loading ipython_console...
2023-03-19 20:31:30,021 [INFO] [spyder.app.mainwindow] -> Registering shortcuts for ipython_console...
2023-03-19 20:31:30,025 [INFO] [spyder.app.mainwindow] -> Loading Layout...
2023-03-19 20:31:30,027 [INFO] [spyder.app.mainwindow] -> Loading layout...
2023-03-19 20:31:30,027 [INFO] [spyder.app.mainwindow] -> Registering shortcuts for layout...
2023-03-19 20:31:30,043 [INFO] [spyder.app.mainwindow] -> Loading Python interpreter...
2023-03-19 20:31:30,048 [INFO] [spyder.app.mainwindow] -> Loading main_interpreter...
2023-03-19 20:31:30,048 [INFO] [spyder.app.mainwindow] -> Registering shortcuts for main_interpreter...
2023-03-19 20:31:30,065 [INFO] [spyder.app.mainwindow] -> Loading Main menus...
2023-03-19 20:31:30,101 [INFO] [spyder.app.mainwindow] -> Loading mainmenu...
2023-03-19 20:31:30,101 [INFO] [spyder.app.mainwindow] -> Registering shortcuts for mainmenu...
2023-03-19 20:31:30,149 [INFO] [spyder.app.mainwindow] -> Loading Online help...
2023-03-19 20:31:30,153 [INFO] [spyder.app.mainwindow] -> Loading onlinehelp...
2023-03-19 20:31:30,165 [INFO] [spyder.app.mainwindow] -> Registering shortcuts for onlinehelp...
2023-03-19 20:31:30,219 [INFO] [spyder.app.mainwindow] -> Loading Outline Explorer...
2023-03-19 20:31:30,224 [INFO] [spyder.app.mainwindow] -> Loading outline_explorer...
2023-03-19 20:31:30,232 [INFO] [spyder.app.mainwindow] -> Registering shortcuts for outline_explorer...
2023-03-19 20:31:30,259 [INFO] [spyder.app.mainwindow] -> Loading Plots...
2023-03-19 20:31:30,263 [INFO] [spyder.app.mainwindow] -> Loading plots...
2023-03-19 20:31:30,271 [INFO] [spyder.app.mainwindow] -> Registering shortcuts for plots...
2023-03-19 20:31:30,279 [INFO] [spyder.app.mainwindow] -> Loading Preferences...
2023-03-19 20:31:30,281 [INFO] [spyder.app.mainwindow] -> Loading preferences...
2023-03-19 20:31:30,282 [INFO] [spyder.app.mainwindow] -> Registering shortcuts for preferences...
2023-03-19 20:31:30,355 [INFO] [spyder.app.mainwindow] -> Loading Profiler...
2023-03-19 20:31:30,359 [INFO] [spyder.app.mainwindow] -> Loading profiler...
2023-03-19 20:31:30,371 [INFO] [spyder.app.mainwindow] -> Registering shortcuts for profiler...
2023-03-19 20:31:30,398 [INFO] [spyder.app.mainwindow] -> Loading Projects...
2023-03-19 20:31:30,403 [INFO] [spyder.app.mainwindow] -> Loading project_explorer...
2023-03-19 20:31:30,411 [INFO] [spyder.app.mainwindow] -> Registering shortcuts for project_explorer...
2023-03-19 20:31:30,443 [INFO] [spyder.app.mainwindow] -> Loading Code Analysis...
2023-03-19 20:31:30,447 [INFO] [spyder.app.mainwindow] -> Loading pylint...
2023-03-19 20:31:30,457 [INFO] [spyder.app.mainwindow] -> Registering shortcuts for pylint...
2023-03-19 20:31:30,502 [INFO] [spyder.app.mainwindow] -> Loading PYTHONPATH manager...
2023-03-19 20:31:30,506 [INFO] [spyder.app.mainwindow] -> Loading pythonpath_manager...
2023-03-19 20:31:30,506 [INFO] [spyder.app.mainwindow] -> Registering shortcuts for pythonpath_manager...
2023-03-19 20:31:30,507 [INFO] [spyder.app.mainwindow] -> Loading Run...
2023-03-19 20:31:30,507 [INFO] [spyder.app.mainwindow] -> Loading run...
2023-03-19 20:31:30,507 [INFO] [spyder.app.mainwindow] -> Registering shortcuts for run...
2023-03-19 20:31:30,508 [INFO] [spyder.app.mainwindow] -> Loading Keyboard shortcuts...
2023-03-19 20:31:30,509 [INFO] [spyder.app.mainwindow] -> Loading shortcuts...
2023-03-19 20:31:30,509 [INFO] [spyder.app.mainwindow] -> Registering shortcuts for shortcuts...
2023-03-19 20:31:30,516 [INFO] [spyder.app.mainwindow] -> Loading Status bar...
2023-03-19 20:31:30,517 [INFO] [spyder.app.mainwindow] -> Loading statusbar...
2023-03-19 20:31:30,518 [INFO] [spyder.app.mainwindow] -> Registering shortcuts for statusbar...
2023-03-19 20:31:30,536 [INFO] [spyder.app.mainwindow] -> Loading Toolbar...
2023-03-19 20:31:30,537 [INFO] [spyder.app.mainwindow] -> Loading toolbar...
2023-03-19 20:31:30,538 [INFO] [spyder.app.mainwindow] -> Registering shortcuts for toolbar...
2023-03-19 20:31:30,545 [INFO] [spyder.app.mainwindow] -> Loading Interactive tours...
2023-03-19 20:31:30,547 [INFO] [spyder.app.mainwindow] -> Loading tours...
2023-03-19 20:31:30,548 [INFO] [spyder.app.mainwindow] -> Registering shortcuts for tours...
2023-03-19 20:31:30,570 [INFO] [spyder.app.mainwindow] -> Loading Variable explorer...
2023-03-19 20:31:30,573 [INFO] [spyder.app.mainwindow] -> Loading variable_explorer...
2023-03-19 20:31:30,580 [INFO] [spyder.app.mainwindow] -> Registering shortcuts for variable_explorer...
2023-03-19 20:31:30,598 [INFO] [spyder.app.mainwindow] -> Loading Working directory...
2023-03-19 20:31:30,710 [INFO] [spyder.app.mainwindow] -> Loading workingdir...
2023-03-19 20:31:30,711 [INFO] [spyder.app.mainwindow] -> Registering shortcuts for workingdir...
2023-03-19 20:31:30,895 [INFO] [spyder.app.mainwindow] -> Loading Terminal...
2023-03-19 20:31:30,905 [INFO] [spyder.app.mainwindow] -> Loading terminal...
2023-03-19 20:31:30,918 [INFO] [spyder.app.mainwindow] -> Registering shortcuts for terminal...
2023-03-19 20:31:30,918 [INFO] [spyder.app.mainwindow] -> Loading old third-party plugins...
2023-03-19 20:31:30,922 [INFO] [spyder.app.mainwindow] -> Creating Menus...
2023-03-19 20:31:30,925 [INFO] [spyder.app.mainwindow] -> Creating toolbars...
2023-03-19 20:31:30,925 [INFO] [spyder.app.mainwindow] -> Setting up main window...
2023-03-19 20:31:30,931 [INFO] [spyder.app.mainwindow] -> Setting up window...
2023-03-19 20:31:31,120 [INFO] [spyder.app.mainwindow] -> *** End of MainWindow setup ***
2023-03-19 20:31:31,286 [INFO] [spyder.plugins.layout.plugin] -> Restoring visible plugins from the previous session
2023-03-19 20:31:31,563 [INFO] [spyder.plugins.ipythonconsole.utils.kernelspec] -> Kernel command: ['/Library/Frameworks/Python.framework/Versions/3.9/bin/python3', '-m', 'spyder_kernels.console', '-f', '{connection_file}']
2023-03-19 20:31:31,657 [INFO] [spyder.plugins.ipythonconsole.utils.kernelspec] -> Kernel command: ['/Library/Frameworks/Python.framework/Versions/3.9/bin/python3', '-m', 'spyder_kernels.console', '-f', '{connection_file}']
2023-03-19 20:31:31,721 [INFO] [spyder.plugins.ipythonconsole.utils.kernelspec] -> Kernel command: ['/Library/Frameworks/Python.framework/Versions/3.9/bin/python3', '-m', 'spyder_kernels.console', '-f', '{connection_file}']
2023-03-19 20:31:32,623 [INFO] [spyder.app.mainwindow] -> Restoring opened files from the previous session
2023-03-19 20:31:32,738 [INFO] [spyder.plugins.completion.providers.languageserver.provider] -> Starting LSP client for python...
2023-03-19 20:31:32,739 [INFO] [spyder.plugins.completion.providers.languageserver.client] -> Starting server: /Applications/Spyder.app/Contents/MacOS/python -m pylsp --host 127.0.0.1 --port 2088 --tcp --check-parent-process --log-file /Users/benjaminlear/.spyder-py3/lsp_logs/server_python_21232.log -v
2023-03-19 20:31:32,751 [INFO] [spyder.plugins.completion.providers.languageserver.client] -> Starting transport for python: /Applications/Spyder.app/Contents/MacOS/python -u /Applications/Spyder.app/Contents/Resources/lib/python3.9/spyder/plugins/completion/providers/languageserver/transport/main.py --folder /Users/benjaminlear/.spyder-py3/lsp_paths/root_path --transport-debug 2 --server-host 127.0.0.1 --server-port 2088  --zmq-in-port 58122 --zmq-out-port 58123 --external-server
Traceback (most recent call last):
  File "/Applications/Spyder.app/Contents/Resources/__boot__.py", line 251, in <module>
    _run()
  File "/Applications/Spyder.app/Contents/Resources/__boot__.py", line 174, in _run
    exec(compile(source, path, "exec"), globals(), globals())
  File "/Applications/Spyder.app/Contents/Resources/spyder", line 3, in <module>
    start.main()
  File "/Applications/Spyder.app/Contents/Resources/lib/python3.9/spyder/app/start.py", line 266, in main
    mainwindow.main(options, args)
  File "/Applications/Spyder.app/Contents/Resources/lib/python3.9/spyder/app/mainwindow.py", line 1822, in main
    mainwindow = create_window(MainWindow, app, splash, options, args)
  File "/Applications/Spyder.app/Contents/Resources/lib/python3.9/spyder/app/utils.py", line 300, in create_window
    main.post_visible_setup()
  File "/Applications/Spyder.app/Contents/Resources/lib/python3.9/spyder/app/mainwindow.py", line 1053, in post_visible_setup
    self.reopen_last_session()
  File "/Applications/Spyder.app/Contents/Resources/lib/python3.9/spyder/app/mainwindow.py", line 1091, in reopen_last_session
    editor.setup_open_files(close_previous_files=False)
  File "/Applications/Spyder.app/Contents/Resources/lib/python3.9/spyder/plugins/editor/plugin.py", line 3474, in setup_open_files
    self.load(filenames[index::-1], goto=clines[index::-1],
  File "/Applications/Spyder.app/Contents/Resources/lib/python3.9/spyder/plugins/editor/plugin.py", line 2273, in load
    finfo = self.editorstacks[0].load(
  File "/Applications/Spyder.app/Contents/Resources/lib/python3.9/spyder/plugins/editor/widgets/editor.py", line 2736, in load
    text, enc = encoding.read(filename)
  File "/Applications/Spyder.app/Contents/Resources/lib/python3.9/spyder/utils/encoding.py", line 308, in read
    text, encoding = decode( open(filename, 'rb').read() )
TimeoutError: [Errno 60] Operation timed out
2023-03-19 20:31:34.011 Spyder[21232:1324052] Launch error
2023-03-19 20:31:34.011 Spyder[21232:1324052] Launch error
See the py2app website for debugging launch issues
Uncaught TypeError: Cannot read property 'fit' of undefined
QProcess: Destroyed while process ("/Applications/Spyder.app/Contents/MacOS/python") is still running.
_distutils_hack/__init__.pyc:18: UserWarning: Distutils was imported before Setuptools, but importing Setuptools also replaces the `distutils` module in `sys.modules`. This may lead to undesirable behaviors or errors. To avoid these issues, avoid using distutils directly, ensure that setuptools is installed in the traditional way (e.g. not an editable install), and/or make sure that setuptools is always imported before distutils.
_distutils_hack/__init__.pyc:33: UserWarning: Setuptools is replacing distutils.
QProcess: Destroyed while process ("/Applications/Spyder.app/Contents/MacOS/python") is still running.
QProcess: Destroyed while process ("/Applications/Spyder.app/Contents/MacOS/python") is still running.

**Comment by mrclary** [comment]
It looks like there may be an issue loading a file in the Editor. Can you try resetting Spyder's configuration? Rename the directory `~/.spyder-py3` to `~/.spyder-py3.bak` and try starting Spyder again from the command-line.

**Comment by ccordoba12** [comment]
@benlear, @mrclary, this is a duplicate of issue #20670, which I already fixed and it'll be part of our next version (5.4.3), to be released in a couple of weeks.

@benlear, a known workaround until that version is out is to start the OneDrive service before starting Spyder.

**Comment by benlear** [comment]
Great!  Thanks so much for the help, and I am sorry I didn't find this on
my own.

**Comment by ccordoba12** [comment]
That's fine, don't worry about it.
